### DISCUSSION TITLE
Adblock

### DISCUSSION THREAD

**Discussion by etherbiswas** [qutebrowser/qutebrowser#7065] [Q&A]
someone please help! I need proper adblock and a plugin system! I love the idea of simply installing extensions to have the broswer do thing i want it to. Thanks in advanced.

**Comment by mschilli87** [comment]
@Ahanaf-Ether: What kind of help are you expecting exactly? There are open issues detailing the progress on both adblocking and extension support. There are even project boards communicationg clearly where we stand and what's the current progress. There is a mailing list and a blog that will inform you once these features land. So unluess you expect somebody to skip the actual work it takes to get what you want by magically snipping the fingers I really do not see what makes you think that bothering close to 200 people with this message could help in any way. The Q&A ist for discussions that start with a question you want answers to. You didn't ask any and every question I can image you might have (*e.g.* Is this planned? When can I expect to see those features? Why does it take so long? What can I do to help?) are already answered in the resources I mentioned above. So please search the available information on the topics, be patient and consider actively supporting development in any fashion that will free time for the people working on those features to make that idea you love a reality instead of wasting their time by making them read and respond to the same comments over and over again. :wink:

**Comment by toofar** [comment]
We don't currently have a system for extensions where you can just press a button and install them, for better or for worse. Like @mschilli87 mentioned, their progress is being tracked around the place ([issue](https://github.com/qutebrowser/qutebrowser/issues/30) and [project](https://github.com/qutebrowser/qutebrowser/projects/3)). But even when that is done they aren't going to be compatible with the existing Chrome and Firefox extensions.
For now we have [userscripts](https://qutebrowser.org/doc/userscripts.html), which can help integrate with external tools. You call them via [spawn](https://qutebrowser.org/doc/help/commands.html#spawn) and hints mostly, there is plenty of documentation about how to use, and create, them. There are some tracked [in the repo](https://github.com/qutebrowser/qutebrowser/tree/master/misc/userscripts).
As for adblock, see point 8 in [the faq](https://qutebrowser.org/doc/faq.html#:~:text=Is%20there%20an%20ad%20blocker). We support an Adblock Plus compatible syntax, minus any cosmetic rules. It's proper enough, but there isn't any fancy UI or anything like that.
